### PR TITLE
Add forwarding timeout option

### LINKED ISSUE #1454: Add timeout option for forwarded request
Currently forwarded messages don’t have a timeout, so if rippled is slow Clio may have many hanging ws connections to rippled. We should add a forwarding timeout option to Clio to fix the issue.

### PR DESCRIPTION
**Pull request by kuznetsss**
Fixes #1454.

### CODE REVIEW

File: src/util/requests/impl/WsConnectionImpl.hpp [hunk 1]
```diff
@@ -92,6 +117,31 @@ class WsConnectionImpl : public WsConnection {
...
+        boost::asio::steady_timer timer{boost::asio::get_associated_executor(cyield), timeout};
+        timer.async_wait([&cancellationSignal](boost::system::error_code errorCode) {
+            if (!errorCode)
+                cancellationSignal.emit(boost::asio::cancellation_type::terminal);
+        });
+        operation(cyield);
+    }
+
+    static boost::system::error_code
+    mapError(boost::system::error_code const ec)
```
Comment (cindyyan317) [inline src/util/requests/impl/WsConnectionImpl.hpp:138]: Do we add 'const' for value parameter?

Comment (kuznetsss) [reply src/util/requests/impl/WsConnectionImpl.hpp:138]: I think it is useful, showing that `ec` will not be changed.

File: src/util/requests/impl/WsConnectionImpl.hpp [hunk 2]
```diff
@@ -51,27 +57,46 @@ class WsConnectionImpl : public WsConnection {
...
-        if (errorCode)
+        auto operation = [&](auto&& token) { ws_.async_write(boost::asio::buffer(message), token); };
+        if (timeout) {
+            withTimeout(operation, yield[errorCode], *timeout);
+        } else {
+            operation(yield[errorCode]);
+        }
+
+        if (errorCode) {
+            errorCode = mapError(errorCode);
```
Comment (cindyyan317) [inline src/util/requests/impl/WsConnectionImpl.hpp:97]: Does the map error only need to happen when withTimeout? My concern is if it shadows any other error?

**Review by cindyyan317 — APPROVED** [review]
LGTM ! 💯

File: src/util/requests/impl/WsConnectionImpl.hpp (hunk 2 above)
Comment (kuznetsss) [reply src/util/requests/impl/WsConnectionImpl.hpp:97]: No, it only to convert `operation_canceled` into `timed_out`.  You can see it at the line 138 in this file.

**Pull request merged**